[Bug] alpha-7ee6809 运行时崩溃

### Verify steps

- [X]  确保你使用的是**本仓库**最新的的 mihomo 或 mihomo Alpha 版本 Ensure you are using the latest version of Mihomo or Mihomo Alpha from **this repository**. 
- [ ]  如果你可以自己 debug 并解决的话，提交 PR 吧 Is this something you can **debug and fix**? Send a pull request! Bug fixes and documentation fixes are welcome. 
- [X]  我已经在 [Issue Tracker](……/) 中找过我要提出的问题 I have searched on the [issue tracker](……/) for a related issue. 
- [X]  我已经使用 Alpha 分支版本测试过，问题依旧存在 I have tested using the dev branch, and the issue still exists. 
- [X]  我已经仔细看过 [Documentation](https://wiki.metacubex.one/) 并无法自行解决问题 I have read the [documentation](https://wiki.metacubex.one/) and was unable to solve the issue. 
- [X]  这是 Mihomo 核心的问题，并非我所使用的 Mihomo 衍生版本（如 OpenMihomo、KoolMihomo 等）的特定问题 This is an issue of the Mihomo core *per se*, not to the derivatives of Mihomo, like OpenMihomo or KoolMihomo. 

### Mihomo version

Mihomo Meta alpha-7ee6809 windows amd64 with go1.21.5 Tue Dec 12 16:14:24 UTC 2023 Use tags: with_gvisor

### What OS are you seeing the problem on?

Windows

### Mihomo config

```yaml
allow-lan: false
bind-address: "*"
find-process-mode: strict
mode: rule
geodata-mode: true
geodata-loader: standard
global-ua: clash.meta
geox-url:
  geoip: "https://fastly.jsdelivr.net/gh/MetaCubeX/meta-rules-dat@release/geoip.dat"
  geosite: "https://fastly.jsdelivr.net/gh/MetaCubeX/meta-rules-dat@release/geosite.dat"
  mmdb: "https://fastly.jsdelivr.net/gh/MetaCubeX/meta-rules-dat@release/geoip.metadb"
log-level: error
ipv6: true
global-client-fingerprint: chrome
keep-alive-interval: 15
tun:
  enable: true
  stack: mixed
  auto-detect-interface: true
  auto-route: true
  endpoint-independent-nat: true
```


### Mihomo log

panic: runtime error: slice bounds out of range [:3] with capacity 1

goroutine 3350 [running]:
github.com/metacubex/mihomo/component/process.convertDOSPath({0xf97433, 0x7})
        github.com/metacubex/mihomo/component/process/process_windows.go:246 +0x272
github.com/metacubex/mihomo/component/process.findProcessName({0xf95486, 0x0?}, {{0x0?, 0xc001c0db00?}, 0xc000114048?}, 0x89)
        github.com/metacubex/mihomo/component/process/process_windows.go:109 +0x196
github.com/metacubex/mihomo/component/process.FindProcessName(...)
        github.com/metacubex/mihomo/component/process/process.go:20
github.com/metacubex/mihomo/tunnel.match(0xc001a9b900)
        github.com/metacubex/mihomo/tunnel/tunnel.go:624 +0x385
github.com/metacubex/mihomo/tunnel.resolveMetadata(0x19d856?)
        github.com/metacubex/mihomo/tunnel/tunnel.go:297 +0x71
github.com/metacubex/mihomo/tunnel.handleUDPConn.func2()
        github.com/metacubex/mihomo/tunnel/tunnel.go:378 +0x155
created by github.com/metacubex/mihomo/tunnel.handleUDPConn in goroutine 8
        github.com/metacubex/mihomo/tunnel/tunnel.go:362 +0x3ac

### Description

其他版本没问题。

--- a/component/process/process_windows.go
+++ b/component/process/process_windows.go
@@ -3,8 +3,6 @@ package process
 import (
 	"fmt"
 	"net/netip"
-	"path/filepath"
-	"strings"
 	"sync"
 	"syscall"
 	"unsafe"
@@ -103,10 +101,6 @@ func findProcessName(network string, ip netip.Addr, srcPort int) (uint32, string
 		return 0, "", err
 	}
 	pp, err := getExecPathFromPID(pid)
-	if err != nil {
-		return 0, "", err
-	}
-	pp, err = convertDOSPath(pp)
 	return 0, pp, err
 }
 
@@ -224,7 +218,7 @@ func getExecPathFromPID(pid uint32) (string, error) {
 	r1, _, err := syscall.SyscallN(
 		queryProcName,
 		uintptr(h),
-		uintptr(1),
+		uintptr(0),
 		uintptr(unsafe.Pointer(&buf[0])),
 		uintptr(unsafe.Pointer(&size)),
 	)
@@ -232,30 +226,4 @@ func getExecPathFromPID(pid uint32) (string, error) {
 		return "", err
 	}
 	return syscall.UTF16ToString(buf[:size]), nil
-}
-
-// modify from https://github.com/shirou/gopsutil/blob/9deadc99147d80f732af3a59e624af73d0143891/internal/common/common_windows.go#L220-L241
-// Convert paths using native DOS format like:
-//
-//	"\Device\HarddiskVolume1\Windows\systemew\file.txt"
-//
-// into:
-//
-//	"C:\Windows\systemew\file.txt"
-func convertDOSPath(p string) (string, error) {
-	rawDrive := strings.Join(strings.Split(p, `\`)[:3], `\`)
-
-	for d := 'A'; d <= 'Z'; d++ {
-		szDeviceName := string(d) + ":"
-		deviceName, err := syscall.UTF16PtrFromString(szDeviceName)
-		if err != nil {
-			return "", err
-		}
-		szTarget := make([]uint16, 512)
-		n, err := windows.QueryDosDevice(deviceName, &szTarget[0], uint32(len(szTarget)))
-		if err == nil && windows.UTF16ToString(szTarget[:n]) == rawDrive {
-			return filepath.Join(szDeviceName, p[len(rawDrive):]), nil
-		}
-	}
-	return p, nil
 }

